### PR TITLE
feat(components): Deprecate components

### LINKED ISSUE #351: Remove use of HtmlElement
Remove `HtmlElement` component in favor of using Emotion's new `as` prop.

### PR DESCRIPTION
**Pull request by connor-baer**
Resolves #351 and addresses #469.

## Purpose

In preparation for v2 of Circuit UI, we're deprecating a few components to give users a head's up of the upcoming changes. 

## Approach and changes

- Remove the `HtmlElement` component (was only used internally)
- Deprecate the `element` prop on the `Heading`, `SubHeading`, `Text`, `List` and `Input` components
- Deprecate the `CreditCardDetails`, `CardNumberInput`, `NameOnCardInput`, `SecurityCodeInput`, and `ExpiryDateInput` components in favor of the [SumUp card widget](https://developer.sumup.com/docs/widgets-card-v2)
- Deprecate the `Markdown` component
- Deprecate the `MaskedInput` component
- Deprecate the `Picture` component
- Deprecate the `SideNav` component
- Rename `ListView` to `CardList`
- Rename `GlobalStyles` to `BaseStyles` deprecate support for custom styles to discourage wrong usage
- Update documentation

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: src/components/RestrictedInput/RestrictedInput.js [hunk 1]
```diff
@@ -78,6 +79,12 @@ RestrictedInput.defaultProps = {
   filteredKeys: []
 };
 
+export function InternalRestrictedInput(props) {
+  deprecate('MaskedInput has been deprecated.');
```
Comment (connor-baer) [inline src/components/RestrictedInput/RestrictedInput.js]: Fix

**Comment by fernandofleury** [comment]
@connor-baer So your idea to remove the `GlobalStyles` is to the enforce the user to actually use `GlobalStyles` from emotion to write their rules?

File: src/components/CardList/index.js [hunk 2]
```diff
@@ -14,21 +14,20 @@
  */
```
Comment (fernandofleury) [inline src/components/CardList/index.js]: @connor-baer You sure this one looks correct? It says GlobalStyles -> CardList

Comment (connor-baer) [reply src/components/CardList/index.js]: I saw that, too. It's just `git` being confused.

**Pull request merged**